app/src/components/pages/ProfessorProfile.tsx

```diff
@@ -1,7 +1,165 @@
+  );
+
+  function getIcon(faculty: Faculty): ReactElement {
+    switch (faculty) {
+      case 'SENG':
+        return <Code />;
+      case 'CSC':
+        return <BubbleChart />;
+      case 'ECE':
+        return <DeveloperBoard />;
+      default:
+        return <DataObject />;
+    }
+  }
+
+  function getBgColor(faculty: Faculty): string {
+    switch (faculty) {
+      case 'SENG':
+        return '#ffc107';
+      case 'CSC':
+        return '#4caf50';
+      case 'ECE':
+        return '#2196f3';
+      default:
+        return '#ff9800';
+    }
+  }
+
+  function getWilling(willingness: number) {
+    if (willingness < 120) {
+      return 'Not willing to teach';
+    }
+    if (willingness < 150) {
+      return 'Somewhat willing to teach';
+    }
+    if (willingness < 170) {
+      return 'Willing to teach';
+    }
+    if (willingness < 200) {
+      return 'Very willing to teach';
+    }
+  }
+
+  return (
+    <Grid display="flex" marginTop={4} justifyContent="center" sx={{ width: '100%' }}>
+      <Card elevation={10} sx={{ width: '70%' }}>
+        <Grid display={'flex'} flexDirection={'row'} margin={4}>
+          <Grid item mr={2}>
+            <Avatar sx={{ bgcolor: '#3f80b5' }} />
+          </Grid>
+          <Grid item>
+            <Typography variant="h4">
+              <b>{professor.username}</b>
+            </Typography>
+          </Grid>
+        </Grid>
+        <Grid display={'flex'} flexDirection={'row'} margin={4}>
+          <Grid item>
+            <Typography variant="body1" color={professor.active ? 'green' : 'red'}>
+              <b>{professor.active ? 'Active' : 'Inactive'}</b>
+            </Typography>
+            <Typography variant="body1">
+              <b>Username: </b> {professor.username}
+            </Typography>
+            <Typography variant="body1">
+              <b>Faculty: </b> {professor.faculty}
+            </Typography>
+            <Typography variant="body1">
+              <b>Role: </b> {professor.role}
+            </Typography>
+          </Grid>
+        </Grid>
+        <Grid display={'flex'} flexDirection={'column'} margin={4}>
+          <Grid item>
+            <Typography variant="h5" marginBottom={2}>
+              <b>Preferences</b>
+            </Typography>
+          </Grid>
+          {professor.preferences.length > 0 ? (
+            professor.preferences?.map((preference: IProfessorPreference) => (
+              <Grid item>
+                <Typography variant="body1">
+                  <b>{preference.id.subject + ' ' + preference.id.code}</b>: {getWilling(preference.preference)}
+                </Typography>
+              </Grid>
+            ))
+          ) : (
+            <Typography variant="body1">No preferences recorded.</Typography>
+          )}
+        </Grid>
+        <Grid display={'flex'} flexDirection={'column'} margin={4}>
+          <Grid item>
+            <Typography variant="h5">
+              <b>Currently Teaching</b>

```

Comment:
Is this going to just be the current semester or for the year?

Reply:
This is a good question, and an important distinction we should decide on. Right now it's just getting all the courses attributed to a teacher but we should narrow it down for sure. What do you think makes more sense? Semester?

Reply:
I added in a check for the start and end date of the course, so it will only display active classes (for the current semester). We can revisit it later.